extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/src/main/kotlin/org/jboss/resteasy/reactive/server/runtime/kotlin/ApplicationCoroutineScope.kt

```diff
@@ -0,0 +1,42 @@
+package org.jboss.resteasy.reactive.server.runtime.kotlin
+
+import io.vertx.core.Context
+import kotlinx.coroutines.*
+import org.eclipse.microprofile.context.ManagedExecutor
+import javax.enterprise.context.ApplicationScoped
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Application wide coroutine scope. Should start and die with the rest of the application, along with child coroutines
+ * (structured concurrency).
+ *
+ * This should be the main interception point to place user specific [CoroutineContext.Element] objects. Things
+ * like activity ids, logger context propagation, tracing, exception handlers, etc.
+ */
+@ApplicationScoped
+class ApplicationCoroutineScope : CoroutineScope, AutoCloseable {
+    override val coroutineContext: CoroutineContext = SupervisorJob()
+
+    override fun close() {
```

Comment:
Why is this class `AutoCloseable`? At which point is `close called?

Reply:
The intention was to signal the container that the bean should be closed on application shutdown. Spring does this automatically and I've seen other frameworks do it as well. Doesn't Quarkus do this on the runtime?
On build time I have `QuarkusBuildCloseablesBuildItem` that seems to do that explicitly. Should I do it that way instead?

Reply:
>  Spring does this automatically and I've seen other frameworks do it as well. Doesn't Quarkus do this on the runtime?

I don't think Quarkus automatically takes `AutoCloseable` into account. There are [lifecycle callbacks](https://quarkus.io/guides/cdi#lifecycle-callbacks) and annotations for [startup and shutdown events](https://quarkus.io/guides/lifecycle).

> On build time I have QuarkusBuildCloseablesBuildItem that seems to do that explicitly. Should I do it that way instead?

Possibly, but I don't know enough about it.

Reply:
You should put a @PreDestory annotation on the method and it will be called when the app is shutdown. QuarkusBuildCloseablesBuildItem is for things that need to be closed after the application is built, its not a runtime thing.

Reply:
Thanks for the clarification. Added the annotation.